aptos-move/framework/aptos-framework/sources/configs/stake.move

```diff
@@ -1606,10 +1606,15 @@ module aptos_framework::stake {
         assert_validator_state(signer::address_of(account), 100, 0, 0, 0, 0);
     }
 
+    #[test_only]
+    public fun set_validator_perf_num_blocks(num_blocks: u64) acquires ValidatorPerformance {
+        borrow_global_mut<ValidatorPerformance>(@aptos_framework).num_blocks = num_blocks;
+    }
+
     #[test_only]
     fun end_epoch() acquires StakePool, StakePoolEvents, TestCoinCapabilities, ValidatorConfig, ValidatorPerformance, ValidatorSet, ValidatorSetConfiguration {
```

Comment:
Why not make this public instead? Do you have use cases for setting a custom num_blocks value in a test?

Reply:
Ok, I tried, but it doesn't work. 
I have a code like this for epoch changes in tests
```
    public entry fun epoch(aptos_framework: &signer) {
        timestamp::update_global_time_for_test(timestamp::now_microseconds() + 1);
        stake::set_validator_num_blocks(1);
        reconfiguration::force_reconfigure(aptos_framework);
    }
```
If I change the second line into `end_epoch()`, it's going to run `stake::on_new_epoch()` twice and fail with div-by-zero. 
If I remove the second line, `reconfiguration::reconfigure_()` is not called, and the state isn't properly updated.

Reply:
Oh I see that makes sense if you force reconfigure.